assets/js/modules/analytics/datastore/report.test.js

```diff
@@ -425,5 +419,72 @@ describe( 'modules/analytics report', () => {
 				expect( isNotGathered ).toBe( false );
 			} );
 		} );
+
+		describe( 'hasZeroData', () => {
+			it( 'should return undefined if getReport or isGatheringData is not resolved yet', async () => {
+				freezeFetch( analyticsRegexp );
+
+				const hasZeroData = registry
+					.select( MODULES_ANALYTICS )
+					.hasZeroData();
+
+				expect( hasZeroData ).toBeUndefined();
+			} );
+
+			it.each( [
+				[ 'an empty array', [ { data: { rows: [] } } ] ],
+				[ 'null', null ],
+			] )(
+				'should return TRUE if report data in isGatheringData OR isZeroReport is %s',
+				async ( _, body ) => {
+					fetchMock.getOnce( analyticsRegexp, { body } );
+
+					const hasZeroData = registry
+						.select( MODULES_ANALYTICS )
+						.hasZeroData();
+
+					expect( hasZeroData ).toBeUndefined();
+
+					await subscribeUntil(
+						registry,
+						() =>
+							registry
+								.select( MODULES_ANALYTICS )
+								.hasZeroData() !== undefined
+					);
+
+					const zeroData = registry
+						.select( MODULES_ANALYTICS )
+						.hasZeroData();
+
+					expect( zeroData ).toBe( true );
+				}
+			);
+
+			it( 'should return FALSE if isGatheringData and isZeroReport return FALSE', async () => {
```

Comment:
Same for these tests as for the above one; I think they're actually two different tests and should be split into two tests.